Source/DafnyCore/Compilers/SinglePassCompiler.cs

```diff
@@ -129,6 +129,10 @@ public abstract class SinglePassCompiler : Plugins.Compiler {
     protected string IntSelect = ",int";
     protected string LambdaExecute = "";
 
+    protected static bool UnicodeChars => UnicodeCharactersOption.Instance.Get(DafnyOptions.O);
```

Comment:
Rename this variable. `UnicodeChars` suggests the variable holds a set/array of characters. But it's actually a boolean. How about something like `UnicodeMode` or `InUnicodeMode` or `FullUnicodeSupport` or `UnicodeSupport``?

Reply:
Completely agree. Perhaps `UnicodeCharEnabled`, to align as closely as possible with the phrasing of the option?

I'm wary of the fact that we're already abusing the term "Unicode" by saying things like "Full unicode support". UTF-16 is already a completely reasonable Unicode encoding that can cover all code points, and we're miles away from "supporting" all concepts from the entire standard, like collation and extended grapheme clusters.

"`--unicode-char`" should only be interpreted to mean "define the `char` type to be more correct and interoperable with other Unicode implementations".